rustls/src/msgs/enums.rs

```diff
@@ -77,7 +77,7 @@ enum_builder! {
     /// from the various RFCs covering TLS, and are listed by IANA.
     /// The `Unknown` item is used when processing unrecognised ordinals.
     @U16
-    pub(crate) enum ExtensionType {
+    pub enum ExtensionType {
```

Comment:
Why is this necessary/useful?

Reply:
Good question, I should have called this out somewhere.

With this changeset the existing `ExtensionType` enum ends up being embedded in the `ClientExtension::EncryptedClientHelloOuterExtensions` extension, and since `ClientExtension` type is `pub` we get a clippy warn without the visibility change:

```
warning: type `msgs::enums::ExtensionType` is more private than the item `msgs::handshake::ClientExtension::EncryptedClientHelloOuterExtensions::0`
   --> rustls/src/msgs/handshake.rs:563:41
    |
563 |     EncryptedClientHelloOuterExtensions(Vec<ExtensionType>),
    |                                         ^^^^^^^^^^^^^^^^^^ field `msgs::handshake::ClientExtension::EncryptedClientHelloOuterExtensions::0` is reachable at visibility `pub`
```